### PR TITLE
Split Jetstream management api from regular api

### PR DESCRIPTION
**Pull request by scottf**
Separate the JetStream Management API from the Publish/Subscribe portion

### CODE REVIEW

File: src/main/java/io/nats/client/JetStreamManagement.java [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+     * publish during reconnect.
+     *
+     * @param subject the subject to send the message to
+     * @param body the message body
+     * @return The publish acknowledgement
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    PublishAck publish(String subject, byte[] body) throws IOException, JetStreamApiException;
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/JetStreamManagement.java:131]: Don't think these should be needed here...

File: src/main/java/io/nats/client/JetStreamManagementOptions.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    private JetStreamManagementOptions(String prefix, Duration requestTimeout) {
+        super(prefix, requestTimeout, false);
+    }
+
+    /**
+     * SubscribeOptions can be created using a Builder. The builder supports chaining and will
+     * create a default set of options if no methods are calls.
+     */
+    public static class Builder {
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/JetStreamManagementOptions.java:34]: Can you subclass from JetstreamOptions.Builder, or if not, encapsulate?

File: src/main/java/io/nats/client/SubscribeOptions.java [hunk 3]
```diff
@@ -104,17 +103,26 @@ public static Builder builder() {
         return new Builder();
     }
 
+    // TODO
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/SubscribeOptions.java:106]: OK at this point, but not sure if you wanted to leave this in for this PR.

File: src/main/java/io/nats/client/JetStreamManagementOptions.java (hunk 2 above)
Comment (scottf) [reply src/main/java/io/nats/client/JetStreamManagementOptions.java:34]: hmmm, maybe

**Pull request merged**